### PR TITLE
Add transcripts to YouTube pages

### PR DESCRIPTION
**Pull request by jacksonh**

### CODE REVIEW

File: packages/content-handler/src/websites/youtube-handler.ts [hunk 1]
```diff
@@ -66,11 +106,12 @@ export class YoutubeHandler extends ContentHandler {
       <iframe width="${width}" height="${height}" src="https://www.youtube.com/embed/${videoId}" title="${title}" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
         <p><a href="${url}" target="_blank">${title}</a></p>
         <p itemscope="" itemprop="author" itemtype="http://schema.org/Person">By <a href="${oembed.author_url}" target="_blank">${authorName}</a></p>
+        <p class='omnivore-youtube-transcript'>
```
Comment (jacksonh) [inline packages/content-handler/src/websites/youtube-handler.ts:109]: One issue is what happens if this is null? I guess just an empty string wouldn't cause any issues here.

File: packages/content-handler/src/websites/youtube-handler.ts [hunk 2]
```diff
@@ -53,6 +84,15 @@ export class YoutubeHandler extends ContentHandler {
     const width = height * ratio
     const authorName = _.escape(oembed.author_name)
 
+    console.log('got video id', videoId)
+    const defaultLanguageCode = await this.getDefaultLanguageCode(videoId)
+
+    const response = await YoutubeTranscript.fetchTranscript(videoId, {
```
Comment (jacksonh) [inline packages/content-handler/src/websites/youtube-handler.ts:90]: Do we know if this can throw?

Comment (sywhb) [reply packages/content-handler/src/websites/youtube-handler.ts]: Added a try-catch block here

File: packages/content-handler/src/websites/youtube-handler.ts (hunk 1 above)
Comment (sywhb) [reply packages/content-handler/src/websites/youtube-handler.ts:115]: Default empty string

**Comment by jacksonh** [comment]
nice, the one thing I'm a bit worried about is the regex to parse the JSON.

**Comment by sywhb** [comment]
> nice, the one thing I'm a bit worried about is the regex to parse the JSON.

Yeah, but I'm not sure about the JSON data structure of the response.
Let me find one

**Comment by jacksonh** [comment]
I think this might fail with this URL but need to test again: https://www.youtube.com/watch?v=KdQHAeAnHmw&t=377s

**Comment by jacksonh** [comment]
Nevermind, it just failed with my code to fetch bookmarks too

**Comment by SilentVoid13** [comment]
Hey there @jacksonh, any update on this? That's a nice feature that I would love to see in Omnivore

**Comment by hientt53** [comment]
I'm really looking forward to this feature from you

**Comment by hanislm** [comment]
Kant wait.

Highlight during read aloud of an article or while watching the the transcription flow of youtube or podcast would also be *chefs kiss*.

**Comment by rilaby** [comment]
Highly awaited feature. 
Are there any particular help needed to be merged in?

**Comment by jacksonh** [comment]
@rilaby I'm going to close this one as I have a more full featured branch i am working on now. Some other upgrades allowed us to simplify this work

**Comment by rilaby** [comment]
May I follow that branch, if you'd be able to share it?

**Pull request closed without merging**